__tests__/search.js

```diff
@@ -35,28 +35,51 @@ class MockVFSServiceProvider extends ServiceProvider {
     ];
   }
 }
+class MySearchAdapter {
+    constructor(core) {
+        this.core = core;
 
+    }
+
+    destroy() {
+    }
+
+    async init() {
+    }
+
+    async search(pattern) {
+        const list = [
+            {
+             path: 'jest:/test.jpg',
+             filename: pattern+'.jpg'   
+            }
+        ];
+        return list;
+    }
+}
 describe('Search', () => {
   let core;
   let search;
-
+  let options = {};
   beforeAll(() => createInstance((c) => {
     c.register(MockVFSServiceProvider);
   }).then(c => core = c));
 
   afterAll(() => core.destroy());
 
   test('#constructor', () => {
-    search = new Search(core);
+    search = new Search(core, options = {
+       adapters: [MySearchAdapter],
+  });
   });
 
   test('#search', () => {
-    return search.search('hello')
-      .then(result => {
-        expect(result).toMatchObject([{
-          path: 'jest:/hello.world',
-          filename: 'hello.world'
-        }]);
+    return search.search('test')
```

Comment:
Why was this rewritten ? The custom adapter doesn't need to contain any pattern matching because the purpose of this test is just to see that if adding another adapter will yield more results.

Just do this:

```js
  test('#search', () => {
    return search.search('hello')
      .then(result => {
        expect(result).toMatchObject([{
          path: 'jest:/hello.world',
          filename: 'hello.world'
        }, {
          // This is based on comment in review
          path: 'custom:/test.jpg',
          filename: 'test.jpg',
        }]);
      });
  });

```

Reply:
> 



> Why was this rewritten ? The custom adapter doesn't need to contain any pattern matching because the purpose of this test is just to see that if adding another adapter will yield more results.
> 
> Just do this:
> 
> ```js
>   test('#search', () => {
>     return search.search('hello')
>       .then(result => {
>         expect(result).toMatchObject([{
>           path: 'jest:/hello.world',
>           filename: 'hello.world'
>         }, {
>           // This is based on comment in review
>           path: 'custom:/test.jpg',
>           filename: 'test.jpg',
>         }]);
>       });
>   });
> ```

Why was this rewritten ? The custom adapter doesn't need to contain any pattern matching because the purpose of this test is just to see that if adding another adapter will yield more results.